sdk/src/azure/core/az_json_token.c

```diff
@@ -325,6 +325,52 @@ AZ_NODISCARD static az_result _az_json_token_get_string_helper(
   return AZ_OK;
 }
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
+    int32_t destination_max_size,
+    int32_t* out_string_length)
+{
+  _az_PRECONDITION_VALID_SPAN(json_string, 1, false);
+  _az_PRECONDITION_NOT_NULL(destination);
+  _az_PRECONDITION(destination_max_size > 0);
+
+  int32_t position = 0;
+  for (int32_t i = 0; i < json_string._internal.size; i++)
+  {
+    uint8_t current_char;
+    current_char = json_string._internal.ptr[i];
+    if (current_char == '\\' && i < json_string._internal.size)
+    {
+      uint8_t next_char = json_string._internal.ptr[i + 1];
+      // check that we have sometig to escape
+      if (_az_is_valid_escaped_character(next_char))
+      {
+        current_char = _az_json_unescape_single_byte(next_char);
+        i++;
```

Comment:
What happens if the input is `"A\\\"\"Z"`? We seem to be skipping past all sequences of backslashes, which isn't correct.

The output in this case should be:
`A\""Z`


Please add test cases.

Reply:
actually you are wrong "A\\\"\"Z" broken into chars is 
0 A
1 \\  -> \     
2 \"  -> "    
3 \"   -> "
4 Z

char 1 is \ char 2 is " which unescaped become \" meaning " 
so the final unquted string is A""Z , quoted is "A\"\"Z"

Reply:
Any test input that has multiple repeating `\` characters will highlight the issue with the current implementation. It is skipping through all sequences of `\` characters. That's what I am trying to point out here.

The two characters `\\` should be transformed into `\`, when unescaped, not ignored.

The test input I am talking about is the following 8 character valid JSON string:
0 A
1 \\
2 \\
3 \\
4 \"
5  \\
6 \"
7 Z

That should be transformed into the following string, once unescaped:
`A\""Z` (5 characters)

Can you share how I am wrong here?